src/include/simeng/RegisterValue.hh

```diff
@@ -26,10 +26,16 @@ class RegisterValue {
    * number of bytes (defaulting to the size of the template type). */
   template <class T,
             typename std::enable_if_t<!std::is_pointer_v<T>, T>* = nullptr>
-  RegisterValue(T value, uint16_t bytes = sizeof(T)) : bytes(bytes) {
+  RegisterValue(T value, uint16_t bytes = sizeof(T), bool relaxFor32 = true) : bytes(bytes) {
+    relaxedFor32bit_ = relaxFor32;
+    std::memset(this->value, 0, MAX_LOCAL_BYTES);
     if (isLocal()) {
       T* view = reinterpret_cast<T*>(this->value);
-      view[0] = value;
+      if (sizeof(T) > bytes) { // e.g. when T is int64 and bytes is 4
```

Comment:
What would be the use case for such an addition? This will cause a truncation during reinterpret_cast. This might work for a specific use case but could cause problems in the future if something like this is allowed.

Why can't you use int32 instead?